File: src/main/java/com/_14ercooper/worldeditor/functions/commands/world/BlockCommand.java
package com._14ercooper.worldeditor.functions.commands.world;

import java.util.List;

import org.bukkit.Material;
import org.bukkit.block.Block;

import com._14ercooper.worldeditor.functions.Function;
import com._14ercooper.worldeditor.functions.commands.InterpreterCommand;
import com._14ercooper.worldeditor.main.Main;

public class BlockCommand extends InterpreterCommand {

    @Override
    public void run(List<String> args, Function function) {
	String[] blocks = args.get(0).split(";");
	int x = (int) function.parseVariable(args.get(1));
	int y = (int) function.parseVariable(args.get(2));
	int z = (int) function.parseVariable(args.get(3));
	Block blk = function.player.getWorld().getBlockAt(x, y, z);
	boolean returnVal = false;
	for (String s : blocks) {
	    Material m = Material.matchMaterial(s);
	    if (!s.contains("#")) {
		if (m == null) {
		    Main.logError("Material " + s + " not found.", function.player);
		}
		else {
		    returnVal = returnVal || m == blk.getType();
		}
	    }
	    else {
		returnVal = returnVal || blk.getType().toString().toLowerCase().contains(s.substring(1).toLowerCase());
	    }
	}
	if (returnVal) function.cmpres = 0;
	else function.cmpres = 1;
    }
}

File: src/main/java/com/_14ercooper/worldeditor/functions/commands/world/BlockCommand.kt
/**
 * This file is part of 14erEdit.
 * 
  * 14erEdit is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * 14erEdit is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with 14erEdit.  If not, see <https://www.gnu.org/licenses/>.
 */

package com._14ercooper.worldeditor.functions.commands.world

import com._14ercooper.worldeditor.functions.Function
import com._14ercooper.worldeditor.functions.commands.InterpreterCommand
import com._14ercooper.worldeditor.main.Main.Companion.logError
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.entity.Player

class BlockCommand : InterpreterCommand() {
    override fun run(args: List<String>, function: Function) {
        val blocks = args[0].split(";").toTypedArray()
        val x = function.parseVariable(args[1]).toInt()
        val y = function.parseVariable(args[2]).toInt()
        val z = function.parseVariable(args[3]).toInt()
        val blk: Block =
            (if (function.player is Player) function.player.world else Bukkit.getServer().worlds[0]).getBlockAt(x, y, z)
        var returnVal = false
        for (s in blocks) {
            val m = Material.matchMaterial(s)
            if (!s.contains("#")) {
                if (m == null) {
                    logError("Material $s not found.", function.player, null)
                } else {
                    returnVal = returnVal || m == blk.type
                }
            } else {
                returnVal = returnVal || blk.type.toString().toLowerCase().contains(s.substring(1).toLowerCase())
            }
        }
        if (returnVal) function.cmpres = 1.0 else function.cmpres = 0.0
    }
}